packages/angular_devkit/core/src/analytics/forwarder.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonObject } from '../json';
+import { Analytics, EventOptions, PageviewOptions, ScreenviewOptions, TimingOptions } from './api';
+
+
+export enum AnalyticsReportKind {
+  Event = 'event',
+  Screenview = 'screenview',
+  Pageview = 'pageview',
+  Timing = 'timing',
+}
+
+export interface AnalyticsReportBase extends JsonObject {
+  kind: AnalyticsReportKind;
+}
+
+export interface AnalyticsReportEvent extends AnalyticsReportBase {
+  kind: AnalyticsReportKind.Event;
+  options: JsonObject & EventOptions;
+  c: string;
+  a: string;
+}
+export interface AnalyticsReportScreenview extends AnalyticsReportBase {
+  kind: AnalyticsReportKind.Screenview;
+  options: JsonObject & ScreenviewOptions;
+  s: string;
+  a: string;
+}
+export interface AnalyticsReportPageview extends AnalyticsReportBase {
+  kind: AnalyticsReportKind.Pageview;
+  options: JsonObject & PageviewOptions;
+  p: string;
+}
+export interface AnalyticsReportTiming extends AnalyticsReportBase {
+  kind: AnalyticsReportKind.Timing;
+  options: JsonObject & TimingOptions;
+  c: string;
+  v: string;
+  t: string | number;
+}
+
+export type AnalyticsReport =
+  AnalyticsReportEvent
+  | AnalyticsReportScreenview
+  | AnalyticsReportPageview
+  | AnalyticsReportTiming
+  ;
+
+/**
+ * A function that can forward analytics along some stream. AnalyticsReport is already a
+ * JsonObject descendant, but we force it here so the user knows it's safe to serialize.
+ */
+export type AnalyticsForwarderFn = (report: JsonObject & AnalyticsReport) => void;
+
+/**
+ * A class that follows the Analytics interface and forward analytic reports (JavaScript objects).
+ * AnalyticsReporter is the counterpart which takes analytic reports and report them to another
+ * Analytics interface.
+ */
+export class AnalyticsForwarder implements Analytics {
```

Comment:
The other two implementations follow the `SomethingAnalytics` naming scheme but this one doesn't. Maybe this should be called `ForwardingAnalytics`?

Reply:
Good name, done.